From f62c2bf7f56695b3cc5599ee24edccf2e1749651 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Wed, 23 Sep 2015 15:05:00 +0530
Subject: [PATCH] USBC_PD: Update power role bit in pd header post PR_SWAP.

This patch will fix the PD header's power role bit not getting
updated based on current power role of the device.

Change-Id: I32e0f23d2aa54b24010ae422f913bf38d7bae6c3
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-18930
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/usb/typec/pd/protocol.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/pd/protocol.c b/drivers/usb/typec/pd/protocol.c
index c3f2225..585f410 100644
--- a/drivers/usb/typec/pd/protocol.c
+++ b/drivers/usb/typec/pd/protocol.c
@@ -80,7 +80,7 @@ static void pd_policy_update_data_role(struct pd_prot *prot,
 	else if (drole == DATA_ROLE_DFP)
 		prot->data_role = PD_DATA_ROLE_DFP;
 
-	queue_work(system_nrt_wq, &prot->role_chng_work);
+	schedule_work(&prot->role_chng_work);
 }
 
 static void pd_policy_update_power_role(struct pd_prot *prot,
@@ -92,9 +92,9 @@ static void pd_policy_update_power_role(struct pd_prot *prot,
 	if (prole == POWER_ROLE_NONE || prole == POWER_ROLE_SINK)
 		prot->pwr_role = PD_POWER_ROLE_CONSUMER;
 	else if (prole == POWER_ROLE_SOURCE)
-		prot->pwr_role = PD_POWER_ROLE_CONSUMER;
+		prot->pwr_role = PD_POWER_ROLE_PROVIDER;
 
-	queue_work(system_nrt_wq, &prot->role_chng_work);
+	schedule_work(&prot->role_chng_work);
 }
 
 static struct prot_msg *prot_alloc_msg(void)
-- 
1.9.1

